doc/source/development/extending.rst

```diff
@@ -208,13 +208,201 @@ will
+  It essential that function signature is compatible with the numpy original.
+  Otherwise,, numpy will ignore it.
+
+  For example, the signature for `np.round` is `np.round(a, decimals=0, out=None)`.
+  if you implement a round function which omits the `out` keyword:
+
+.. code-block:: python
+
+    def round(self, decimals=0):
+        pass
+
+
+  numpy will ignore it. The following will work however:
+
+.. code-block:: python
+
+    def round(self, decimals=0, **kwds):
+        pass
+
+An alternative approach to implementing individual functions, is to override
+`__getattr__` in your ExtensionArray, and to intercept requests for method
+names which you wish to support (such as `round`). For most functions,
+you can return a dynamically generated function, which simply calls
+the numpy function on your existing backing numeric array, wraps
+the result in your ExtensionArray, and returns it. This approach can
+reduce boilerplate significantly, but you do have to maintain a whitelist,
+and may require more than one case, based on signature.
+
+
+A third possible approach, is to use the `__array_function__`
+mechanism introduced by [NEP18](https://www.numpy.org/neps/nep-0018-array-function-protocol.html).
+This is an opt-in mechanism in numpy 1.16 (by setting an environment variable), and
+is enabled by default starting with numpy 1.17. As of 1.17 it is still considered
+experimental, and its design is actively being revised. We will not discuss it further
+here, but it is certainly possible to make use of it to achieve the same goal. Your
+mileage may vary.
+
+.. important::
+    Implementing `__array_function__` is not a substitute for implementing `__array_ufunc__`.
+    The `__array_function__` mechanism complements (and to a degree copies) the`__array_ufunc__`
+    mechanism, by providing the same flexibility for non-ufuncs.
+
+.. important::
+    `__array_function__` is an "all-in" solution. That means that if you cannot mix it with
+    explicit implementations for some methods and using `__array_function__` for some.
+    If you both `__array_function__` and also provide an implementation of `round`, numpy
+    will invoke `__array_function__` for all the operations in the specification, **including**
+     `round`.
+
+With this overview in hand, you hopefully have the necessary information in order
+to develop rich, full-featured ExtensionArrays that seamlessly plug in to pandas.
+
+.. important::
+    You are not required to provide implementations for the full complement of Series
+    operations in your ExtensionArray. In fact, some of them may not even make sense
+    within your context. You may also choose to ass implementations incrementally,
+    as the need arises.
+
+    TBD: should we have a standard way of signalling not supported instead of a
+    random AttributeError exception being thrown.
+
+.. important::
+
+The above description currently leads the state of the code considerably. Many Series
```

Comment:
What does this first sentence mean?

You're not under the `.. important` block.

And I don't think having three `important` blocks in a row is good. Then everything becomes unimportant.

Reply:
It means I'm uncomfortable about documenting how it works, when I know full well that many of the series methods are currently broken for EA, and don't behave as described. 

I removed the important note, and added another sentence to the close, stating EA are still being worked on, so please report bugs.